From d3b684d3ec3f6455f85811693bf0120a40e56b35 Mon Sep 17 00:00:00 2001
From: Evan Chen <evanxinchen@google.com>
Date: Tue, 23 Sep 2025 18:17:07 +0000
Subject: [PATCH 28/55] Validate displayName for AssociationRequest

validate the length of displayName in the constructor, setDisplayName
and the builder

Test: cts
Flag: EXEMPT BUGFIX
Bug: 443742829
Cherrypick-From: https://googleplex-android-review.googlesource.com/q/commit:110f3e5843d87e4a994eb07160064ca3552355dd
Merged-In: I41dc3804c35328a5968716f502c4320a58b75158
Change-Id: I41dc3804c35328a5968716f502c4320a58b75158
---
 .../android/companion/AssociationRequest.java     | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/core/java/android/companion/AssociationRequest.java b/core/java/android/companion/AssociationRequest.java
index fc6fe845b..f2c8a35ff 100644
--- a/core/java/android/companion/AssociationRequest.java
+++ b/core/java/android/companion/AssociationRequest.java
@@ -318,6 +318,7 @@ public final class AssociationRequest implements Parcelable {
             boolean forceConfirmation,
             boolean skipRoleGrant,
             @Nullable Icon deviceIcon) {
+        validateDisplayName(displayName);
         mSingleDevice = singleDevice;
         mDeviceFilters = requireNonNull(deviceFilters);
         mDeviceProfile = deviceProfile;
@@ -431,6 +432,7 @@ public final class AssociationRequest implements Parcelable {
 
     /** @hide */
     public void setDisplayName(CharSequence displayName) {
+        validateDisplayName(displayName);
         mDisplayName = displayName;
     }
 
@@ -522,11 +524,7 @@ public final class AssociationRequest implements Parcelable {
         public Builder setDisplayName(@NonNull CharSequence displayName) {
             checkNotUsed();
             mDisplayName = requireNonNull(displayName);
-            if (displayName.length() > DISPLAY_NAME_LENGTH_LIMIT) {
-                throw new IllegalArgumentException("Length of the display name must be at most "
-                        + DISPLAY_NAME_LENGTH_LIMIT + " characters");
-            }
-
+            validateDisplayName(displayName);
             return this;
         }
 
@@ -821,4 +819,11 @@ public final class AssociationRequest implements Parcelable {
             return new AssociationRequest(in);
         }
     };
+
+    private static void validateDisplayName(@Nullable CharSequence displayName) {
+        if (displayName != null && displayName.length() > DISPLAY_NAME_LENGTH_LIMIT) {
+            throw new IllegalArgumentException("Length of the display name must be at most "
+                    + DISPLAY_NAME_LENGTH_LIMIT + " characters");
+        }
+    }
 }
-- 
2.52.0

